tests/integration/ha/test_backups.py

```diff
@@ -0,0 +1,568 @@
+async def test_backup_cluster(
+    ops_test: OpsTest,
+    c_writes: ContinuousWrites,
+    c_writes_runner,
+    cloud_configs,
+    cloud_credentials,
+    cloud_name,
+) -> None:
+    """Runs the backup process whilst writing to the cluster into 'noisy-index'."""
+    global global_backup_id
+
+    app = (await app_name(ops_test)) or APP_NAME
+    leader_id = await get_leader_unit_id(ops_test)
+    unit_ip = await get_leader_unit_ip(ops_test)
+    config = cloud_configs[cloud_name]
+
+    logger.info(f"Syncing credentials for {cloud_name}")
+    await _configure_s3(ops_test, config, cloud_credentials[cloud_name], app)
+
+    logger.info("Creating backup")
+    global_backup_id = await backup_cluster(
+        ops_test,
+        leader_id,
+    )
+    assert global_backup_id > 0
+    if cloud_name not in backups_by_cloud:
+        backups_by_cloud[cloud_name] = []
+    backups_by_cloud[cloud_name].append(global_backup_id)
+
+    # Comparing the number of docs without stopping c_writes
+    initial_count = await index_docs_count(ops_test, app, unit_ip, ContinuousWrites.INDEX_NAME)
+    time.sleep(5)
+    count = await index_docs_count(ops_test, app, unit_ip, ContinuousWrites.INDEX_NAME)
+    assert count > initial_count
+
+    # continuous writes checks
+    await assert_continuous_writes_consistency(ops_test, c_writes, app)
+
+
+@pytest.mark.parametrize(
+    "cloud_name",
+    [
+        (pytest.param("microceph", marks=pytest.mark.group("microceph"))),
+        (pytest.param("aws", marks=pytest.mark.group("aws"))),
+    ],
+)
+@pytest.mark.abort_on_fail
+async def test_restore_cluster(
+    ops_test: OpsTest, cloud_configs, cloud_credentials, cloud_name
+) -> None:
+    """Restores the cluster and tries to search for index."""
+    global global_backup_id
+
+    unit_ip = await get_leader_unit_ip(ops_test)
+    app = (await app_name(ops_test)) or APP_NAME
+    leader_id = await get_leader_unit_id(ops_test)
+    config = cloud_configs[cloud_name]
+
+    logger.info(f"Syncing credentials for {cloud_name}")
+    await _configure_s3(ops_test, config, cloud_credentials[cloud_name], app)
+
+    logger.info("Restoring backup")
+    assert await restore_cluster(
+        ops_test,
+        global_backup_id,  # backup_id
+        unit_ip,
+        leader_id,
+    )
+    assert await start_and_check_continuous_writes(ops_test, unit_ip, app)
+
+
+@pytest.mark.parametrize(
+    "cloud_name",
+    [
+        (pytest.param("microceph", marks=pytest.mark.group("microceph"))),
+        (pytest.param("aws", marks=pytest.mark.group("aws"))),
+    ],
+)
+@pytest.mark.abort_on_fail
+async def test_restore_cluster_after_app_destroyed(
+    ops_test: OpsTest, cloud_configs, cloud_credentials, cloud_name
+) -> None:
+    """Deletes the entire OpenSearch cluster and redeploys from scratch.
+
+    Restores the backup and then checks if the same TEST_BACKUP_INDEX is there.
+    """
+    global global_backup_id
+
+    app = (await app_name(ops_test)) or APP_NAME
+
```

Comment:
we need to get and store the count of documents of the target backup here

Reply:
it is grabbed from the opensearch APIs.

Reply:
Added a global variable for that: `cwrites_backup_doc_count`. This way, when we do the DR test, we run against each of the backups we have knowledge of and double check with the original # of docs present inthis global variable.

We need a global variable b/c backups are executed on separated tests and then we restore each on the DR test.